### PR TITLE
fix(prepare-and-finalize-artifacts): replace oras pull with use

### PR DESCRIPTION
**Pull request by HarelErlich**
Select artifact at runtime and persist choice to a file, then invoke `build-trusted-artifacts use` to pull instead of `oras pull`. This restores support for sha256-only inputs and fixes the "failed to resolve sha256:..." error without changing pipeline behavior.

Signed-off-by: Harel Erlich herlich@redhat.com

## Checklist before requesting a review
- [ ] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [x] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

### CODE REVIEW

File: tasks/managed/prepare-and-finalize-artifacts/prepare-and-finalize-artifacts.yaml [hunk 1]
```diff
@@ -78,17 +79,36 @@ spec:
       script: |
         #!/usr/bin/env bash
         set -euxo pipefail
-
         if [ -n "$(params.rpmDataArtifact)" ]; then
-          ARTIFACT_TO_PULL_WITH_PREFIX="$(params.rpmDataArtifact)"
+          ARTIFACT_TO_USE="$(params.rpmDataArtifact)"
```
Comment (elenagerman) [inline tasks/managed/prepare-and-finalize-artifacts/prepare-and-finalize-artifacts.yaml:86]: Maybe it's worth adding a validation to ensure at least one artifact is provided.
something like:
```suggestion
          ARTIFACT_TO_USE="$(params.rpmDataArtifact)"
        elif [ -n "$(params.baseDataArtifact)" ]; then
          ARTIFACT_TO_USE="$(params.baseDataArtifact)"
        else
          echo "Error: No artifact provided. Either baseDataArtifact or rpmDataArtifact must be specified"
          exit 1
        fi
```

**Comment by elenagerman** [comment]
In addition, I would extend the unit test coverage a little.
you already have a 'task with rpm' verification,
we can add 'task without rpm' and 'base selected' tests. WDYT?

**Review by enriquebelarte — APPROVED** [review]

File: tasks/managed/prepare-and-finalize-artifacts/prepare-and-finalize-artifacts.yaml (hunk 1 above)
Comment (HarelErlich) [reply tasks/managed/prepare-and-finalize-artifacts/prepare-and-finalize-artifacts.yaml:86]: I added it, thanks for the suggestion.

**Comment by HarelErlich** [comment]
@elenagerman Yes, you’re right, i added those tests as well.

File: tasks/managed/prepare-and-finalize-artifacts/tests/test-rpm-artifact.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        - name: baseDataArtifact
+          value: "oci:mock.registry/test-repo/base-artifact@sha256:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"
+        - name: rpmDataArtifact
+          value: "oci:mock.registry/test-repo/rpm-artifact@sha256:abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+        - name: trustedArtifactsDebug
+          value: ""
+        - name: taskGitUrl
+          value: $(params.taskGitUrl)
+        - name: taskGitRevision
+          value: $(params.taskGitRevision)
```
Comment (seanconroy2021) [inline tasks/managed/prepare-and-finalize-artifacts/tests/test-rpm-artifact.yaml]: nit - missing new line

File: tasks/managed/prepare-and-finalize-artifacts/tests/test-no-artifacts.yaml [hunk 3]
```diff
@@ -2,10 +2,13 @@
...
 metadata:
-  name: test-prepare-and-finalize-artifacts
+  name: test-prepare-and-finalize-artifacts-no-artifacts
+  annotations:
+    test/assert-task-failure: "run-task"
 spec:
   description: |
-    Run the prepare-and-finalize-artifacts task and verify its steps are mocked.
+    Test the prepare-and-finalize-artifacts task when no artifacts are provided.
+    This should fail during the use step.
```
Comment (seanconroy2021) [inline tasks/managed/prepare-and-finalize-artifacts/tests/test-no-artifacts.yaml:11]: Will this fail at select-artifact step then?  As baseDataArtifact and rpmDataArtifact are empty.

File: tasks/managed/prepare-and-finalize-artifacts/tests/test-base-artifact-only.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        - name: baseDataArtifact
+          value: "oci:mock.registry/test-repo/base-artifact@sha256:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"
+        - name: rpmDataArtifact
+          value: ""
+        - name: trustedArtifactsDebug
+          value: ""
+        - name: taskGitUrl
+          value: $(params.taskGitUrl)
+        - name: taskGitRevision
+          value: $(params.taskGitRevision)
```
Comment (seanconroy2021) [inline tasks/managed/prepare-and-finalize-artifacts/tests/test-base-artifact-only.yaml]: nit - new line

**Comment by seanconroy2021** [comment]
Would it be worth adding a test for when only the RPM artifact is provided (base is empty)?

File: tasks/managed/prepare-and-finalize-artifacts/tests/test-no-artifacts.yaml (hunk 3 above)
Comment (HarelErlich) [reply tasks/managed/prepare-and-finalize-artifacts/tests/test-no-artifacts.yaml:11]: It’s supposed to fail, but the test passes because I added these lines

  annotations:
    test/assert-task-failure: "run-task"

File: tasks/managed/prepare-and-finalize-artifacts/tests/test-rpm-artifact.yaml (hunk 2 above)
Comment (HarelErlich) [reply tasks/managed/prepare-and-finalize-artifacts/tests/test-rpm-artifact.yaml]: Fixed

File: tasks/managed/prepare-and-finalize-artifacts/tests/test-base-artifact-only.yaml (hunk 4 above)
Comment (HarelErlich) [reply tasks/managed/prepare-and-finalize-artifacts/tests/test-base-artifact-only.yaml]: fixed

File: tasks/managed/prepare-and-finalize-artifacts/tests/test-no-artifacts.yaml (hunk 3 above)
Comment (seanconroy2021) [reply tasks/managed/prepare-and-finalize-artifacts/tests/test-no-artifacts.yaml:11]: Yeah, I understand, but it's failing because baseDataArtifact and rpmDataArtifact are empty, right? Just checking  I understand it correctly.

Comment (HarelErlich) [reply tasks/managed/prepare-and-finalize-artifacts/tests/test-no-artifacts.yaml:11]: Yes, exactly

**Comment by HarelErlich** [comment]
@seanconroy2021 I added the test for only RPM

**Review by seanconroy2021 — APPROVED** [review]
LGTM :)
